src/vs/workbench/contrib/notebook/common/notebookCommon.ts

```diff
@@ -646,22 +646,32 @@ export namespace CellUri {
 		});
 	}
 
-	export function parseCellOutputUri(uri: URI): { notebook: URI; outputId?: string } | undefined {
+	export function parseCellOutputUri(uri: URI): { notebook: URI; openIn: string; outputId?: string; cellFragment?: string; outputIndex?: number } | undefined {
```

Comment:
Instead of returning the `cellFragment`, it might be simpler to return the handle here.